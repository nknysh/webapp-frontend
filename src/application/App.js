/**
 * Copyright Â© 2018 Gigster, Inc. All Rights Reserved.
 *
 * This software and/or related documentation, is protected by copyright and
 * other intellectual property laws, and Gigster and/or its licensors, as
 * applicable, own all right, title and interest in and to its content, and all
 * derivatives, translations, adaptations and combinations of the foregoing. You
 * are not permitted to copy, distribute, transmit, store, display, perform,
 * reproduce, publish, license, create derivative works from, transfer, sell, or
 * make any other use of this software and/or related documentation unless you
 * have entered into a written agreement with Gigster regarding such usage. You
 * agree that all such usage of the software and/or related documentation shall
 * be subject to the terms and conditions of such written agreement, including
 * all applicable license restrictions.
 */

// Libraries
import React from 'react';
import { connect } from 'react-redux';

// App
import { getUserFromToken } from 'actions/auth';
import { AppRouter, getQuery } from '../router';

// Styles
import './App.css';

class App extends React.Component {
  getTokenParam = () => {
    if (!window) {
      return null;
    }

    const query = getQuery(window.location) || {};
    return query.token;
  }

  componentWillMount() {
    // TODO(mark): Probably not the best place but we don't have sagas.
    const tokenParam = this.getTokenParam();
    const token = tokenParam || localStorage.getItem('authToken');

    if (token) {
      this.props.getUserFromToken({ token });
    }
  }

  render() {
    return <AppRouter />;
  }
}

export default connect(undefined, { getUserFromToken })(App);
