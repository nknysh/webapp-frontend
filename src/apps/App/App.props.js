import PropTypes from 'prop-types';

import { noop } from 'utils';

export const propTypes = {
  setToken: PropTypes.func,
  isAppVersionDeprecated: PropTypes.bool,
};

export const defaultProps = {
  setToken: noop,
  isAppVersionDeprecated: false,
};
