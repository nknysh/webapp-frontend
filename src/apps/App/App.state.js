import { connect } from 'react-redux';
import { pipe } from 'ramda';
import { setToken } from 'store/modules/auth/actions';
import { resetStatuses } from 'store/common/actions';
import { getCurrentUser } from 'store/modules/auth/selectors';

export const mapStateToProps = state => ({
  currentUser: getCurrentUser(state),
});

export const mapDispatchToProps = dispatch => ({
  setToken: pipe(
    setToken,
    dispatch
  ),
  resetStatuses: pipe(
    resetStatuses,
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
);
