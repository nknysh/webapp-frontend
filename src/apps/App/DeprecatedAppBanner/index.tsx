import React from 'react';
import styled  from 'styled-components';

const DeprecatedAppBannerComponent = () =>  (
  <div>
    This version of app is deprecated. In order to avoid getting errors, please, reload page to get the latest version.
  </div>
);

const DeprecatedAppBanner = styled(DeprecatedAppBannerComponent)`
  width: 100%;
  background-color: yellow;
`;

export default DeprecatedAppBanner;
