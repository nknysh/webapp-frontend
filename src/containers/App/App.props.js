import PropTypes from 'prop-types';

export const propTypes = {
  getUserFromToken: PropTypes.func,
};
