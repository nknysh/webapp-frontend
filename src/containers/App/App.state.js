import { connect } from 'react-redux';
import { pipe } from 'ramda';
import { getUserFromToken } from 'actions/auth';

export const mapStateToProps = () => ({});

export const mapDispatchToProps = dispatch => ({
  getUserFromToken: pipe(
    getUserFromToken,
    dispatch
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
);
