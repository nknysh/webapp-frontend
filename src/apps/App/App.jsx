import React, { Suspense, useEffect } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import { compose, prop } from 'ramda';
import { ErrorBoundary, Loader } from '@pure-escapes/webapp-ui-components';

import { getAppRoutes } from 'routing';

import { Layout } from 'components';
import { useScrollToTop, useEffectBoundary } from 'effects';
import { withUser } from 'hoc';

import { propTypes, defaultProps } from './App.props';
import connect from './App.state';
import { FastSearchContainerConnected } from 'containers/FastSearch';
import { RatesImportContainerConnected } from 'containers/RatesImport';
import { AllotmentsImportContainerConnected } from 'containers/AllotmentsImport';
import { OfferRouting } from 'containers/OfferRouting';
import { AuthRoute } from 'containers/AuthRoute';
import { EUserType } from 'services/BackendApi';

export const App = ({
  location: { pathname },
  user,
  resetStatuses,
  pageChange,
  bootstrapAppRequestAction,
  isAppVersionDeprecated,
}) => {
  // Scroll to top on path change
  useScrollToTop(pathname);

  useEffect(() => {
    bootstrapAppRequestAction();
  }, [bootstrapAppRequestAction]);

  useEffectBoundary(() => {
    // Trigger page change action on pathname change
    pageChange(pathname);

    // Reset statuses on page change
    resetStatuses();
  }, [pathname]);

  return (
    <Layout isAppVersionDeprecated={isAppVersionDeprecated}>
      <Suspense fallback={<Loader />}>
        <Switch>
          <Route path="/search/beta" exact component={FastSearchContainerConnected} />
          <Route path="/offers" component={OfferRouting} />
          <AuthRoute allow={[EUserType.RL, EUserType.ADMIN]} exact path="/rates/import" component={RatesImportContainerConnected} />
          <AuthRoute allow={[EUserType.RL, EUserType.ADMIN]} exact path="/allotments/import" component={AllotmentsImportContainerConnected} />
          {...getAppRoutes(prop('type', user))}
        </Switch>
      </Suspense>
    </Layout>
  );
};

App.propTypes = propTypes;
App.defaultProps = defaultProps;

export default compose(ErrorBoundary, withRouter, withUser, connect)(App);
