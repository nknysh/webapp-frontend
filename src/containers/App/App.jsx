import React, { PureComponent } from 'react';
import { Switch, BrowserRouter } from 'react-router-dom';
import { compose } from 'ramda';

import routes from 'routing';
import { ErrorBoundary } from 'hoc/ErrorBoundary';
import { getToken } from 'utils/auth';
import { windowExists } from 'utils/window';

import { propTypes } from './App.props';
import connect from './App.state';
import './App.css';

export class App extends PureComponent {
  componentDidMount() {
    const { getUserFromToken } = this.props;
    const token = getToken(windowExists);

    if (token) {
      getUserFromToken(token);
    }
  }

  render() {
    return (
      <BrowserRouter>
        <Switch>{routes}</Switch>
      </BrowserRouter>
    );
  }
}

App.propTypes = propTypes;

export default compose(
  ErrorBoundary,
  connect
)(App);
