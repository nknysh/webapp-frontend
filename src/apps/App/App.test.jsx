import React from 'react';

import { App } from './App';

const getShallow = props => shallow(<App location={{ pathname: '/' }} getUserToken={() => {}} {...props} />);

describe('<App />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      const component = getShallow();
      expect(component).toMatchSnapshot();
    });
  });
});
