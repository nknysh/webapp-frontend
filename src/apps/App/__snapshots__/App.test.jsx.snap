// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render matches snapshot 1`] = `
<withRouter(withUser(Layout))>
  <Suspense
    fallback={
      <ForwardRef
        theme={
          Object {
            "animations": Object {
              "defaultTransition": "ease-in-out 0.25s all",
            },
            "backgrounds": Object {
              "default": "#FFFFFF",
              "defaultOpacity": "rgba(248,248,248,0.85)",
              "light": "#F8F8F8",
              "secondary": "#F8F8F8",
              "whiteOpacity": "rgba(255,255,255, 0.9)",
            },
            "borders": Object {
              "default": "#E0E0E0",
              "medium": "#E9E8E9",
              "normal": "#EAEAEA",
              "range": "#816850",
              "secondary": "#E1E1E1",
            },
            "boxShadows": Object {
              "default": "4px 4px 8px 0 rgba(0,0,0,0.1)",
            },
            "breakpointSizes": Object {
              "desktop": 1280,
              "tablet": 700,
            },
            "breakpoints": Object {
              "desktop": [Function],
              "tablet": [Function],
            },
            "colors": Object {
              "aqua": "#D4F3EE",
              "black": "#373C46",
              "black-darker": "#333333",
              "black-light": "#999999",
              "black-secondary": "#9F9F9F",
              "gold": "#A18265",
              "gold-border": "#816850",
              "gold-dark": "#382A21",
              "gold-light": "#9D9591",
              "gold-neutral": "#736A65",
              "gold-opacity": "rgba(161,130,101, 0.75)",
              "gray": "#EAEAEA",
              "gray-dark": "#E0E0E0",
              "gray-faded": "#9d9591",
              "gray-label": "#736A65",
              "gray-light": "#F8F8F8",
              "gray-medium": "#E9E8E9",
              "gray-neutral": "#a18265",
              "gray-opacity-1": "rgba(0,0,0,0.35)",
              "gray-secondary": "#E1E1E1",
              "green": "#56d1bf",
              "light-blue": "#F3F9FB",
              "marine": "#56D1BF",
              "red-fade": "#FD5656",
              "white": "#FFFFFF",
              "white-opacity-1": "rgba(248,248,248,0.85)",
              "white-opacity-2": "rgba(255,255,255, 0.9)",
              "whiteish": "#F8F8F8",
              "yellow": "#f5a623",
            },
            "fonts": Object {
              "bold": 600,
              "bolder": 800,
              "defaultFont": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
              "headingFont": "NoeDisplay, \\"Open Sans\\", sans-serif",
              "letterSpacing": Object {
                "medium": 0.5,
                "mid": 0.46,
                "whole": 1,
              },
              "light": 300,
              "lighter": 200,
              "normal": "normal",
              "sizes": Object {
                "big": 18,
                "bigger": 22,
                "default": 12,
                "less": 13,
                "link": 11,
                "mid": 16,
                "normal": 14,
              },
            },
            "muiTheme": Object {
              "backgrounds": Object {
                "default": "#FFFFFF",
                "defaultOpacity": "rgba(248,248,248,0.85)",
                "light": "#F8F8F8",
                "secondary": "#F8F8F8",
                "whiteOpacity": "rgba(255,255,255, 0.9)",
              },
              "borders": Object {
                "default": "#E0E0E0",
                "medium": "#E9E8E9",
                "normal": "#EAEAEA",
                "range": "#816850",
                "secondary": "#E1E1E1",
              },
              "breakpoints": Object {
                "between": [Function],
                "down": [Function],
                "keys": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
                "only": [Function],
                "up": [Function],
                "values": Object {
                  "lg": 1280,
                  "md": 960,
                  "sm": 600,
                  "xl": 1920,
                  "xs": 0,
                },
                "width": [Function],
              },
              "direction": "ltr",
              "fonts": Object {
                "bold": 600,
                "bolder": 800,
                "defaultFont": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                "headingFont": "NoeDisplay, \\"Open Sans\\", sans-serif",
                "letterSpacing": Object {
                  "medium": 0.5,
                  "mid": 0.46,
                  "whole": 1,
                },
                "light": 300,
                "lighter": 200,
                "normal": "normal",
                "sizes": Object {
                  "big": 18,
                  "bigger": 22,
                  "default": 12,
                  "less": 13,
                  "link": 11,
                  "mid": 16,
                  "normal": 14,
                },
              },
              "mixins": Object {
                "gutters": [Function],
                "toolbar": Object {
                  "@media (min-width:0px) and (orientation: landscape)": Object {
                    "minHeight": 48,
                  },
                  "@media (min-width:600px)": Object {
                    "minHeight": 64,
                  },
                  "minHeight": 56,
                },
              },
              "overrides": Object {
                "MuiButton": Object {
                  "label": Object {
                    "color": "#FFFFFF",
                    "fontSize": 12,
                    "letterSpacing": "2px !important",
                    "paddingRight": 10,
                  },
                  "root": Object {
                    "backgroundColor": "#A18265",
                    "borderRadius": 0,
                  },
                },
                "Typography": Object {
                  "body2": Object {
                    "fontSize": 18,
                  },
                },
              },
              "palette": Object {
                "action": Object {
                  "active": "rgba(0, 0, 0, 0.54)",
                  "disabled": "rgba(0, 0, 0, 0.26)",
                  "disabledBackground": "rgba(0, 0, 0, 0.12)",
                  "hover": "rgba(0, 0, 0, 0.08)",
                  "hoverOpacity": 0.08,
                  "selected": "rgba(0, 0, 0, 0.14)",
                },
                "augmentColor": [Function],
                "background": Object {
                  "default": "#fafafa",
                  "paper": "#fff",
                },
                "common": Object {
                  "black": "#000",
                  "white": "#fff",
                },
                "contrastThreshold": 3,
                "divider": "rgba(0, 0, 0, 0.12)",
                "error": Object {
                  "contrastText": "#fff",
                  "dark": "#d32f2f",
                  "light": "#e57373",
                  "main": "#f44336",
                },
                "getContrastText": [Function],
                "grey": Object {
                  "100": "#f5f5f5",
                  "200": "#eeeeee",
                  "300": "#e0e0e0",
                  "400": "#bdbdbd",
                  "50": "#fafafa",
                  "500": "#9e9e9e",
                  "600": "#757575",
                  "700": "#616161",
                  "800": "#424242",
                  "900": "#212121",
                  "A100": "#d5d5d5",
                  "A200": "#aaaaaa",
                  "A400": "#303030",
                  "A700": "#616161",
                },
                "primary": Object {
                  "contrastText": "#fff",
                  "dark": "rgb(112, 91, 70)",
                  "light": "rgb(179, 155, 131)",
                  "main": "#A18265",
                },
                "secondary": Object {
                  "contrastText": "#fff",
                  "dark": "rgb(39, 29, 23)",
                  "light": "rgb(95, 84, 77)",
                  "main": "#382A21",
                },
                "text": Object {
                  "disabled": "rgba(0, 0, 0, 0.38)",
                  "hint": "rgba(0, 0, 0, 0.38)",
                  "primary": "rgba(0, 0, 0, 0.87)",
                  "secondary": "rgba(0, 0, 0, 0.54)",
                },
                "tonalOffset": 0.2,
                "type": "light",
              },
              "props": Object {},
              "shadows": Array [
                "none",
                "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
                "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
                "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
                "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
                "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
                "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
                "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
                "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
                "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
                "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
                "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
                "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
                "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
                "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
                "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
                "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
                "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
                "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
                "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
                "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
                "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
                "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
                "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
                "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
              ],
              "shape": Object {
                "borderRadius": 4,
              },
              "spacing": [Function],
              "transitions": Object {
                "create": [Function],
                "duration": Object {
                  "complex": 375,
                  "enteringScreen": 225,
                  "leavingScreen": 195,
                  "short": 250,
                  "shorter": 0.5,
                  "shortest": 150,
                  "standard": 300,
                },
                "easing": Object {
                  "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
                  "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
                  "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
                  "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
                },
                "getAutoHeightDuration": [Function],
              },
              "typography": Object {
                "body1": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "1rem",
                  "fontWeight": 400,
                  "lineHeight": 1.5,
                },
                "body2": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "0.875rem",
                  "fontWeight": 400,
                  "lineHeight": 1.43,
                },
                "button": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "0.875rem",
                  "fontWeight": 500,
                  "lineHeight": 1.75,
                  "textTransform": "uppercase",
                },
                "caption": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "0.75rem",
                  "fontWeight": 400,
                  "lineHeight": 1.66,
                },
                "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                "fontSize": 14,
                "fontWeightBold": 700,
                "fontWeightLight": 300,
                "fontWeightMedium": 500,
                "fontWeightRegular": 400,
                "h1": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "6rem",
                  "fontWeight": 300,
                  "lineHeight": 1,
                },
                "h2": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "3.75rem",
                  "fontWeight": 300,
                  "lineHeight": 1,
                },
                "h3": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "3rem",
                  "fontWeight": 400,
                  "lineHeight": 1.04,
                },
                "h4": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "2.125rem",
                  "fontWeight": 400,
                  "lineHeight": 1.17,
                },
                "h5": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "1.5rem",
                  "fontWeight": 400,
                  "lineHeight": 1.33,
                },
                "h6": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "1.25rem",
                  "fontWeight": 500,
                  "lineHeight": 1.6,
                },
                "htmlFontSize": 16,
                "overline": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "0.75rem",
                  "fontWeight": 400,
                  "lineHeight": 2.66,
                  "textTransform": "uppercase",
                },
                "pxToRem": [Function],
                "round": [Function],
                "subtitle1": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "1rem",
                  "fontWeight": 400,
                  "lineHeight": 1.75,
                },
                "subtitle2": Object {
                  "fontFamily": "HurmeGeometricSans2, \\"Open Sans\\", sans-serif",
                  "fontSize": "0.875rem",
                  "fontWeight": 500,
                  "lineHeight": 1.57,
                },
              },
              "zIndex": Object {
                "appBar": 1100,
                "drawer": 1200,
                "mobileStepper": 1000,
                "modal": 1300,
                "snackbar": 1400,
                "speedDial": 1050,
                "tooltip": 1500,
              },
            },
            "palette": Object {
              "cancelled": "#a18265",
              "confirmed": "#56d1bf",
              "error": "red",
              "light": "#9D9591",
              "link": Object {
                "active": "#A18265",
                "default": "#382A21",
              },
              "listSeparatorColor": "#A18265",
              "navigation": "#F8F8F8",
              "neutral": "#736A65",
              "ok": "green",
              "opacity": 0.35,
              "pending": "#9d9591",
              "primary": "#A18265",
              "primaryOpacity": "rgba(161,130,101, 0.75)",
              "requested": "#f5a623",
              "secondary": "#382A21",
              "sectionLabel": "#736A65",
              "selected": "#D4F3EE",
              "warning": "#f5a623",
            },
            "sizes": Object {
              "borderRadius": 1,
              "headerSizes": Object {
                "mobile": 70,
                "tablet": 120,
              },
            },
            "spacing": Object {
              "gutter": 10,
              "unit": "px",
            },
          }
        }
      />
    }
  >
    <Switch>
      <Route
        auth={true}
        authCheckIgnore={true}
        authComponent={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        exact={true}
        key="5f160202a8dcaaabeadf9bee291c623a7d3f1a1c"
        name="Root Path"
        path="/"
        render={[Function]}
      />
      <Route
        key="2ee726a5e790bd03f0d2000777a30a1eaa355525"
        name="Login"
        path="/login"
        render={[Function]}
      />
      <Route
        ignore={true}
        key="389f64b2402428d0673fc85de311e5583f5b534a"
        name="Logout"
        path="/logout"
        render={[Function]}
      />
      <Route
        key="6dfe9ea18159fe3caf34d7cd34a21dac9e876f83"
        name="Create Account"
        path="/sign-up"
        render={[Function]}
      />
      <Route
        exact={true}
        key="303c6af8abda09ba1b1564a0efa89a9b982ead62"
        name="Password Reset"
        path="/password-reset"
        render={[Function]}
      />
      <Route
        exact={true}
        key="de6820d6f6cd78cb111a78ee07887aa7aa12c9a3"
        name="Password Reset"
        path="/password-reset/:token"
        render={[Function]}
      />
      <Route
        auth={true}
        key="17c782e7d4e39ebecc43aaf235a17a41bbf43b22"
        name="Search page"
        path="/search"
        render={[Function]}
      />
      <Route
        auth={true}
        exact={true}
        key="2ed872f49357b36ccc01eeae870296cad458e6d8"
        name="Hotel"
        path="/hotels/:id"
        render={[Function]}
      />
      <Route
        auth={true}
        exact={true}
        key="59ddfc228b73df48eb18c4906b4fb5b8772c7dbb"
        name="Hotel"
        path="/hotels/:id/:stage/:complete?"
        render={[Function]}
      />
      <Route
        auth={true}
        key="195978ee34e10dca100ed8c1df4ab91f99247f07"
        name="Proposal page"
        path="/proposals/:id/:stage?"
        render={[Function]}
      />
      <Route
        auth={true}
        key="3696bdeab0bf9acf2e4f7841de1cb8abca9ff8f2"
        name="Booking Details"
        path="/bookings/:id/:stage?"
        render={[Function]}
      />
      <Route
        auth={true}
        key="8fdf46ed087281dd3b50712d360ef51138a968b5"
        name="User"
        path="/settings/:section?"
        render={[Function]}
      />
      <Route
        exact={true}
        key="f8523930459db6784973f5c9dde9dba62842916d"
        name="Content Pages"
        path="/:pageId"
        render={[Function]}
      />
      <Route
        key="323217f643c3e3f1fe7532e72ac01bb0748c97be"
        name="Not Found"
        render={[Function]}
      />
    </Switch>
  </Suspense>
</withRouter(withUser(Layout))>
`;
